src/containers/PostDetail/PostDetail.js

```diff
@@ -32,28 +36,60 @@ class PostDetail extends React.Component {
 
   componentDidMount() {
     if (this.__getPostProperty("id")) {
-      this.__fetchPost();
+      this.props.fetchActivePost(this.__getPostProperty("id"));
     }
   }
 
-  __fetchPost() {
-    postsApi.fetchPost(this.__getPostProperty('id'))
-      .then(post => {
-        this.__changePostProperties({
-          title: post.title,
-          body: post.body,
-          authorId: post.userId,
-        });
-      })
-      .catch(console.log);
+  componentWillReceiveProps(nextProps) {
+    if (this.__isPostFetchSuccess(nextProps)) {
+      this.__changePostProperties({...nextProps.activePost.item});
+    } else if (this.__isPostFetchFailure(nextProps)) {
+      this.__displayPostFetchErrorMessage();
```

Comment:
merge all error messages into one functionality and make it global accessible fro all components. If there will be request for raise an error it will be easier to use it